phoenix5-spark/src/main/scala/org/apache/phoenix/spark/DefaultSource.scala

```diff
@@ -17,44 +17,6 @@
  */
 package org.apache.phoenix.spark
 
-import org.apache.spark.sql.sources.{BaseRelation, CreatableRelationProvider, RelationProvider}
-import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
+import org.apache.phoenix.spark.datasource.v2.PhoenixDataSource
 
-@deprecated("Use the DataSource V2 API implementation (see PhoenixDataSource)")
-class DefaultSource extends RelationProvider with CreatableRelationProvider {
-
-  // Override 'RelationProvider.createRelation', this enables DataFrame.load()
-  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]): BaseRelation = {
-    verifyParameters(parameters)
-
-    new PhoenixRelation(
-      parameters("table"),
-      parameters("zkUrl"),
-      parameters.contains("dateAsTimestamp")
-    )(sqlContext)
-  }
-
-  // Override 'CreatableRelationProvider.createRelation', this enables DataFrame.save()
-  override def createRelation(sqlContext: SQLContext, mode: SaveMode,
-                              parameters: Map[String, String], data: DataFrame): BaseRelation = {
-
-    if (!mode.equals(SaveMode.Overwrite)) {
```

Comment:
Is there an advatage to keeping this check ? 
Do we have tests that check for it ?